internal/router/fieldinclude.go

```diff
@@ -27,11 +28,21 @@ func (u *FieldUpdater) UpdateField(fieldName string, val interface{}) {
 	u.fields[fieldName] = val
 }
 
+// UpdateSearch requests that search records are updated.
+func (u *FieldUpdater) UpdateSearch() {
+	u.search = true
+}
+
 // A FieldUpdateFunc is used to update a metadata document for the
 // given id. For each field in fields, it should set that field to
 // its corresponding value in the metadata document.
 type FieldUpdateFunc func(id *charm.Reference, fields map[string]interface{}) error
 
+// A FieldUpdateSearchFunc is used to update a search document for the
+// given id. For each field in fields, it should set that field to
+// its corresponding value in the search document.
+type FieldUpdateSearchFunc func(id *charm.Reference, fields map[string]interface{}) error
```

Comment:
Another possibility here might be to add an "updateSearch" bool parameter to FieldUpdateFunc, which potentially gives the update function a little more information to play with.